packages/relic/lib/src/headers/typed/headers/accept_language_header.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:relic/src/headers/extension/string_list_extensions.dart';
+import 'package:relic/src/headers/typed/base/typed_header.dart';
+
+/// A class representing the HTTP Accept-Language header.
+///
+/// This header specifies the natural languages that are preferred in the response.
+class AcceptLanguageHeader extends TypedHeader {
+  /// The list of languages that are accepted.
+  final List<LanguageQuality>? languages;
+
+  /// A boolean value indicating whether the Accept-Language header is a wildcard.
+  final bool isWildcard;
+
+  /// Constructs an instance of [AcceptLanguageHeader] with the given languages.
+  const AcceptLanguageHeader.languages({required this.languages})
+      : isWildcard = false;
+
+  /// Constructs an instance of [AcceptLanguageHeader] with a wildcard language.
+  const AcceptLanguageHeader.wildcard()
+      : languages = null,
+        isWildcard = true;
+
+  /// Parses the Accept-Language header value and returns an [AcceptLanguageHeader] instance.
+  factory AcceptLanguageHeader.parse(List<String> values) {
+    var splitValues = values.splitTrimAndFilterUnique();
+
+    if (splitValues.isEmpty) {
+      throw FormatException('Value cannot be empty');
+    }
+
+    if (splitValues.length == 1 && splitValues.first == '*') {
+      return AcceptLanguageHeader.wildcard();
+    }
+
+    if (splitValues.length > 1 && splitValues.contains('*')) {
+      throw FormatException('Wildcard (*) cannot be used with other values');
```

Comment:
If my request handler never readers the `Accept-Language` header, a server using this framework will still reject requests if they carry an invalid `Accept-Language` header.

Is that wise? (I'm actually not sure, hence, why I'm asking :rofl: )

What do other server frameworks do here?

Reply:
Hi @jonasfj,  
We have a `strict` flag for headers. When true, any invalid header rejects the request, ensuring valid headers. If false, invalid headers are added to the `failedHeadersToParse` map in `headers.dart`.

Reply:
`strict` mode is our own contraption and can be configured per route in the server. It allows a developer always to reject any request that has invalid headers. The downside is that the header parser will need to parse all headers received when the request reaches the server.

Most other server frameworks do not supply such a feature but require this to be implemented as a middleware if wanted. However, since we provide types for all headers, it was pretty simple to supply the feature with the framework. But with that said, this is an optional feature that the developer can enable at will.

Reply:
disclaimer: I don't have any answers here, so feel free to ignore my drive by comments :rofl: 

I can see it being attractive to have certain headers be `strict`.
Examples:
 * Ignoring a malformed `Content-Type`, `Content-Length`, `Host` could be bad.
 * Ignoring a malformed `Referer`, `Accept-Language` or `Accept` header, might be more reasonable. Indeed rejecting a request over malformed `Accept-Language` header might defeat the point of content-negotiation (especially, if the request handler never reads the header).

Reply:
@jonasfj a new issue is created for this https://github.com/serverpod/serverpod/issues/3034
Thank you for your feedback!